core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/function/NumberFunction.java

```diff
@@ -44,7 +45,10 @@ public void beginAggregate() {
         public void update(StructuredRecord record) {
           Number val = record.get(fieldName);
           if (val != null) {
-            updateDouble(val.doubleValue());
+            Double updateDouble = updateDouble(val.doubleValue());
+            if (val.doubleValue() == updateDouble) {
```

Comment:
I'm having trouble understanding this.  Is it to try and fit all AggregateFunctions into RecordAggregateFunctions?  Think we need some refactoring rather than trying to force things that don't fit together.